spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayMethodMetadata.java

```diff
@@ -30,37 +31,42 @@
  *
  * @author Oleg Zhurakousky
  * @author Gary Russell
+ * @author Artem Bilan
+ *
  * @since 2.0
  */
 public class GatewayMethodMetadata {
 
-	private volatile String payloadExpression;
-
-	private volatile String requestChannelName;
+	private final Map<String, Expression> headerExpressions = new HashMap<>();
 
-	private volatile String replyChannelName;
+	private Expression payloadExpression;
 
-	private volatile String requestTimeout;
+	private String requestChannelName;
 
-	private volatile String replyTimeout;
+	private String replyChannelName;
 
-	private volatile Map<String, Expression> headerExpressions = new HashMap<String, Expression>();
+	private String requestTimeout;
 
+	private String replyTimeout;
 
-	public String getPayloadExpression() {
+	@Nullable
+	public Expression getPayloadExpression() {
 		return this.payloadExpression;
 	}
 
-	public void setPayloadExpression(String payloadExpression) {
+	public void setPayloadExpression(Expression payloadExpression) {
```

Comment:
Breaking change?

Reply:
I realized that, but this is about a new `5.2` generation and there is a little chance that this class is used somewhere outside of the framework.
Plus a workaround is easy as well.
Also we can't have overloaded version with `String` because XML parser with its Java Beans introspection is going to fail.

Reply:
Ok. Agree.